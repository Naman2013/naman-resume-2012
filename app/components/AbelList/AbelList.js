import React from 'react';
import PropTypes from 'prop-types';

const AbelList = ({ list }) => (
  <ul>
    { list.map(item => <li>{item}</li>) }
  </ul>
);

AbelList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
};

AbelList.defaultProps = {
  list: [],
};

export default AbelList;
