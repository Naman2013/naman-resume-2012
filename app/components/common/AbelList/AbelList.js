import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { DeviceContext } from '../../../providers/DeviceProvider';
import style from './AbelList.style';

const AbelList = ({ list }) => (
  <Fragment>
    <DeviceContext.Consumer>
      {
        (context) => {
          console.log(context);
          return (
            <ul>
              { list.map(item => <li>{item}</li>) }
            </ul>
          );
        }
      }

    </DeviceContext.Consumer>
    <style jsx>{style}</style>
  </Fragment>
);

AbelList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
};

AbelList.defaultProps = {
  list: [],
};

export default AbelList;
