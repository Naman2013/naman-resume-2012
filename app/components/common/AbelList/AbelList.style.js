import css from 'styled-jsx/css';
import { primaryFont } from '../../../styles/variables/fonts';

export default css`
  ul {
    font-family: ${primaryFont};
    list-style-type: none;
    margin: 0;
    padding: 0;
  }

  li {
    border-bottom: 1px solid aqua;
  }
`;
