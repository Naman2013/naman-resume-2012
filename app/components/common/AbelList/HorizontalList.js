import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './HorizontalList.style';

const HorizontalList = ({ list, theme, iconList }) => (
  <Fragment>
    <ul style={theme}>
      {list.map((entry, i) => (
        <li key={`horizontal-list-${entry}`}>
          {' '}
          {iconList[i] && <img className="bullet-image" src={iconList[i]} />}
          {entry}
        </li>
      ))}
    </ul>
    <style jsx>{style}</style>
  </Fragment>
);

HorizontalList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  theme: PropTypes.shape({}),
  iconList: PropTypes.arrayOf(PropTypes.string),
};

HorizontalList.defaultProps = {
  theme: {},
  iconList: [],
};

export default HorizontalList;
