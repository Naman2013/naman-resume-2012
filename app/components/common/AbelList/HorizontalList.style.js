import css from 'styled-jsx/css';
import { astronaut, glitter } from '../../../styles/variables/colors_tiles_v4';
import { primaryFont } from '../../../styles/variables/fonts';

const FONT_SIZE = 11;
const TOTAL_HEIGHT = 70;
const VERTICAL_PADDING = (TOTAL_HEIGHT - FONT_SIZE) / 2;

export default css`
  ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
    display: flex;
    font-family: ${primaryFont};
    color: ${astronaut};
    text-transform: uppercase;
    border-top: 1px solid ${glitter};
    border-bottom: 1px solid ${glitter};
  }

  li {
    flex-grow: 1;
    font-weight: bold;
    padding: ${VERTICAL_PADDING}px 0;
    padding-left: 15px;
    font-size: ${FONT_SIZE}px;
    border-right: 1px solid ${glitter};
    letter-spacing: 0.4px;
  }

  li:last-child {
    border-right: none;
  }

  .bullet-image {
    margin-right: 10px;
  }

  .list-container {
    display: flex;
    float: left;
  }
`;
