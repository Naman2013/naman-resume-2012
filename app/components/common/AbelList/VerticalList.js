import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './VerticalList.style';

const VerticalList = ({ list, iconList }) => (
  <Fragment>
    <ul>
      {list.map((entry, i) => (
        <li key={`vertical-list-${entry}`}>
          {iconList[i] && <img className="bullet-image" src={iconList[i]} />}{' '}
          {entry}
        </li>
      ))}
    </ul>
    <style jsx>{style}</style>
  </Fragment>
);

VerticalList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.string),
  iconList: PropTypes.arrayOf(PropTypes.string),
};

VerticalList.defaultProps = {
  list: [],
  iconList: [],
};

export default VerticalList;
