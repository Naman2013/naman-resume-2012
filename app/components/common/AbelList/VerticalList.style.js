import css from 'styled-jsx/css';
import { astronaut, glitter } from '../../../styles/variables/colors_tiles_v4';
import { primaryFont } from '../../../styles/variables/fonts';

export default css`
  ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
  }

  li {
    text-transform: uppercase;
    font-size: 11px;
    font-weight: bold;
    letter-spacing: 0.6px;
    padding: 20px 0;
    border-bottom: 1px solid ${glitter};
    font-family: ${primaryFont};
    color: ${astronaut};
  }

  .bullet-image {
    margin-right: 10px;
  }
`;
