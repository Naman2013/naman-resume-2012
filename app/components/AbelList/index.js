import AbelList from './AbelList';

export default AbelList;
