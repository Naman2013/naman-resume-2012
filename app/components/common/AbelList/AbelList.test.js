import React from 'react';
import { shallow } from 'enzyme';
import AbelList from './AbelList';

describe('AbelList', () => {
  const sampleList = ['Contains 10 objects', 'category guide', 'beginners and up'];
  const shallowWrapper = shallow(<AbelList list={sampleList} />);

  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
