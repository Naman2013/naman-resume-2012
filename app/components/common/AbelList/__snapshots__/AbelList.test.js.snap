// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbelList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AbelList
    list={
      Array [
        "Contains 10 objects",
        "category guide",
        "beginners and up",
      ]
    }
    theme={
      Object {
        "horizontalList": Object {},
        "verticalList": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED>
          [Function]
        </UNDEFINED>,
        <JSXStyle
          css={
            Array [
              "ul.jsx-1298830369{font-family:brandon-grotesque, sans-serif;list-style-type:none;margin:0;padding:0;}",
              "li.jsx-1298830369{border-bottom:1px solid aqua;}",
            ]
          }
          styleId="1298830369"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": [Function],
        },
        "ref": null,
        "rendered": [Function],
        "type": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_defaultValue": undefined,
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "ul.jsx-1298830369{font-family:brandon-grotesque, sans-serif;list-style-type:none;margin:0;padding:0;}",
            "li.jsx-1298830369{border-bottom:1px solid aqua;}",
          ],
          "styleId": "1298830369",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED>
            [Function]
          </UNDEFINED>,
          <JSXStyle
            css={
              Array [
                "ul.jsx-1298830369{font-family:brandon-grotesque, sans-serif;list-style-type:none;margin:0;padding:0;}",
                "li.jsx-1298830369{border-bottom:1px solid aqua;}",
              ]
            }
            styleId="1298830369"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": [Function],
          },
          "ref": null,
          "rendered": [Function],
          "type": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": [Circular],
            "Provider": Object {
              "$$typeof": Symbol(react.provider),
              "_context": [Circular],
            },
            "_calculateChangedBits": null,
            "_changedBits": 0,
            "_changedBits2": 0,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
            "_defaultValue": undefined,
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "ul.jsx-1298830369{font-family:brandon-grotesque, sans-serif;list-style-type:none;margin:0;padding:0;}",
              "li.jsx-1298830369{border-bottom:1px solid aqua;}",
            ],
            "styleId": "1298830369",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
